### ISSUE TITLE
Proxyswitcher is closed under Windows 10

### ISSUE THREAD

**Issue by fgiesecke** [mwiedemeyer/ProxySwitcher#10]
When my PC is going in Standby, proxyswitcher crashes. I had the same problem many times.
In the eventlog I found the following two errors

>Anwendung: ProxySwitcher.exe
Frameworkversion: v4.0.30319
Beschreibung: Der Prozess wurde aufgrund einer unbehandelten Ausnahme beendet.
Ausnahmeinformationen: System.ComponentModel.Win32Exception
Stapel:
   bei Hardcodet.Wpf.TaskbarNotification.TaskbarIcon.CreateTaskbarIcon()
   bei Hardcodet.Wpf.TaskbarNotification.Interop.WindowMessageSink.OnWindowMessageReceived(IntPtr, UInt32, UInt32, UInt32)
   bei MS.Win32.UnsafeNativeMethods.IntGetMessageW(System.Windows.Interop.MSG ByRef, System.Runtime.InteropServices.HandleRef, Int32, Int32)
   bei MS.Win32.UnsafeNativeMethods.IntGetMessageW(System.Windows.Interop.MSG ByRef, System.Runtime.InteropServices.HandleRef, Int32, Int32)
   bei MS.Win32.UnsafeNativeMethods.GetMessageW(System.Windows.Interop.MSG ByRef, System.Runtime.InteropServices.HandleRef, Int32, Int32)
   bei System.Windows.Threading.Dispatcher.GetMessage(System.Windows.Interop.MSG ByRef, IntPtr, Int32, Int32)
   bei System.Windows.Threading.Dispatcher.PushFrameImpl(System.Windows.Threading.DispatcherFrame)
   bei System.Windows.Application.RunDispatcher(System.Object)
   bei System.Windows.Application.RunInternal(System.Windows.Window)
   bei ProxySwitcher.App.Main()

   
-------------------------------------------------
   
>Name der fehlerhaften Anwendung: ProxySwitcher.exe, Version: 3.6.1.322, Zeitstempel: 0x4f6b956a
Name des fehlerhaften Moduls: KERNELBASE.dll, Version: 10.0.10240.16384, Zeitstempel: 0x559f38c3
Ausnahmecode: 0xe0434352
Fehleroffset: 0x000000000002a1c8
ID des fehlerhaften Prozesses: 0x14e8
Startzeit der fehlerhaften Anwendung: 0x01d0f9ad9c6dc018
Pfad der fehlerhaften Anwendung: C:\Program Files (x86)\Proxy Switcher\ProxySwitcher.exe
Pfad des fehlerhaften Moduls: C:\WINDOWS\system32\KERNELBASE.dll
Berichtskennung: 8f37192c-a056-4564-b660-c99e42060dff
Vollständiger Name des fehlerhaften Pakets: 
Anwendungs-ID, die relativ zum fehlerhaften Paket ist:

**Comment by carstenschuette** [comment]
I get the same exception when computer comes back online from standby/hibernation.

```
Application: ProxySwitcher.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.ComponentModel.Win32Exception
Stack:
   at Hardcodet.Wpf.TaskbarNotification.TaskbarIcon.CreateTaskbarIcon()
   at Hardcodet.Wpf.TaskbarNotification.Interop.WindowMessageSink.OnWindowMessageReceived(IntPtr, UInt32, UInt32, UInt32)
   at MS.Win32.UnsafeNativeMethods.IntGetMessageW(System.Windows.Interop.MSG ByRef, System.Runtime.InteropServices.HandleRef, Int32, Int32)
   at MS.Win32.UnsafeNativeMethods.IntGetMessageW(System.Windows.Interop.MSG ByRef, System.Runtime.InteropServices.HandleRef, Int32, Int32)
   at MS.Win32.UnsafeNativeMethods.GetMessageW(System.Windows.Interop.MSG ByRef, System.Runtime.InteropServices.HandleRef, Int32, Int32)
   at System.Windows.Threading.Dispatcher.GetMessage(System.Windows.Interop.MSG ByRef, IntPtr, Int32, Int32)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(System.Windows.Threading.DispatcherFrame)
   at System.Windows.Application.RunDispatcher(System.Object)
   at System.Windows.Application.RunInternal(System.Windows.Window)
   at ProxySwitcher.App.Main()
```
```
Name der fehlerhaften Anwendung: ProxySwitcher.exe, Version: 3.6.2.423, Zeitstempel: 0x56223f17
Name des fehlerhaften Moduls: KERNELBASE.dll, Version: 10.0.10240.16384, Zeitstempel: 0x559f38c3
Ausnahmecode: 0xe0434352
Fehleroffset: 0x000000000002a1c8
ID des fehlerhaften Prozesses: 0xafc
Startzeit der fehlerhaften Anwendung: 0x01d10a47f03304b9
Pfad der fehlerhaften Anwendung: C:\Program Files (x86)\Proxy Switcher\ProxySwitcher.exe
Pfad des fehlerhaften Moduls: C:\WINDOWS\system32\KERNELBASE.dll
Berichtskennung: 959843e5-5f7f-47bf-9d1f-8c9ca87deff4
Vollständiger Name des fehlerhaften Pakets: 
Anwendungs-ID, die relativ zum fehlerhaften Paket ist: 
```

**Comment by mwiedemeyer** [comment]
It seems to be an issue with the [WpfNotifyIcon](https://www.nuget.org/packages/Hardcodet.NotifyIcon.Wpf/) I'm using. Maybe it should be replaced with the WinForms NotifiyIcon, because actually ProxySwitcher is not using any special feature from the WPF NotifiyIcon.

Someone interested in replacing the component with anything else?

**Comment by whedro** [comment]
HI, 
I was same problem .... but after install W10 upgrade (Threshold2) I have one day and lot of sleep modes without crash ...

**Comment by whedro** [comment]
Still working .. :)

**Comment by carstenschuette** [comment]
Correct. Seems to be fixed in Windows 10 version 1511 (TH2).

**Comment by fgiesecke** [comment]
Hm strange...
on my surface pro it is not yet fixed although version 1511 is running fine. My Laptop has no problems.

**Comment by carstenschuette** [comment]
Crashed again today on my PC :-(

**Comment by keeneye** [comment]
I am still having this problem on Windows 10 version 1511

**Comment by mcrustk2** [comment]
Hi,

I have it 1/2 working on Windows 10 switching between HOME (disable Internet Options Proxy) and WORK (should re-tick use Proxy in Internet Options). The auto-detect and HOME work fine (removes proxy info completely), when it detects WORK it doesn't actually 'TICK' the "Use Proxy Server for your LAN" checkbox even though the proxy info is visible again 'although grayed out'. Any ideas how I can get it to actually check the checkbox. Using Win10 1151.

Thanks,

**Comment by HolgerSchade** [comment]
Hi,

I had the same issue that ProxySwitcher crashed when recovering from sleep mode under Windows 10. I took the source and recompiled it with the current  "Hardcodet.Wpf.TaskbarNotification.dll". This dll is embedded in Version 1.0.8 of Wpf.NotifyIcon of Hardcodet. The dll in this package however has a 1.0.5 version number.

The recompile of the core ProxySwitcher worked without problems under VisualStudio 2015. However the setup projects (.vdproj-Projects) are not supported anymore.

Since I recompiled with the newer dll - ProxySwitcher worked flawlessly for about 1 week now.
